roles/openshift_health_checker/openshift_checks/disk_availability.py

```diff
@@ -93,22 +89,14 @@ def run(self):
 
         return {}
 
-    @staticmethod
-    def free_bytes(path, ansible_mounts):
+    def free_bytes(self, path):
         """Return the size available in path based on ansible_mounts."""
-        mount_point = path
-        # arbitry value to prevent an infinite loop, in the unlike case that '/'
-        # is not in ansible_mounts.
-        max_depth = 32
-        while mount_point not in ansible_mounts and max_depth > 0:
-            mount_point = os.path.dirname(mount_point)
-            max_depth -= 1
-
+        mount = self.find_ansible_mount(path)
         try:
-            free_bytes = ansible_mounts[mount_point]['size_available']
+            return mount['size_available']
         except KeyError:
-            known_mounts = ', '.join('"{}"'.format(mount) for mount in sorted(ansible_mounts)) or 'none'
-            msg = 'Unable to determine disk availability for "{}". Known mount points: {}.'
-            raise OpenShiftCheckException(msg.format(path, known_mounts))
-
-        return free_bytes
+            raise OpenShiftCheckException(
+                'Unable to retrieve disk availability for "{}" due to a bug.\n'
```

Comment:
I agree, I would like the user to know what's going on at this level of detail. I get a little lazy when it seems like a really unlikely thing to encounter and complicated to explain accurately if it did happen (because if this is broken, then enough has changed that you probably need to re-evaluate your assumptions), and so stick to simple "something went wrong, and it's not your fault" messages. But you're right, a little further thought can help craft a more helpful error message, both for the user and for the developer.